packages/twenty-server/src/engine/metadata-modules/remote-server/remote-table/remote-table.service.ts

```diff
@@ -438,4 +453,32 @@ export class RemoteTableService {
       }
     }
   }
+
+  private getDistantTablesWithPendingUpdates(
+    schemaPendingUpdates: { [tablename: string]: DistantTableUpdate[] },
+    distantTablesWithStatus: {
+      name: string;
+      schema: string;
+      status: RemoteTableStatus;
+    }[],
+    remoteServerSchema: string,
+  ) {
+    const distantTablesWithUpdates = distantTablesWithStatus.map((table) => ({
+      ...table,
+      schemaPendingUpdates: schemaPendingUpdates[table.name] || [],
+    }));
+
+    const deletedTables = Object.entries(schemaPendingUpdates)
```

Comment:
Not sure of this part. Currently we add deletedTables only when we ask for schema updates, which is a bit confusing